src/mementoasplugin.cpp

```diff
@@ -0,0 +1,325 @@
+                                                                                                                               "1.2.826.0.1.1578918.9.8.1.5");
+    _cass_comm_monitor = new CommunicationMonitor(new Alarm(alarm_manager,
+                                                            "memento",
+                                                            AlarmDef::MEMENTO_AS_CASSANDRA_COMM_ERROR,
+                                                            AlarmDef::CRITICAL),
+                                                  "Memento",
+                                                  "Memcached");
+
+    // We need the address family for the CassandraResolver
+    int af = AF_INET;
+    struct in6_addr dummy_addr;
+    if (inet_pton(AF_INET6, opt.local_host.c_str(), &dummy_addr) == 1)
+    {
+      TRC_DEBUG("Local host is an IPv6 address");
+      af = AF_INET6;
+    }
+
+    // Default to a 30s blacklist/graylist duration and port 9160
+    _cass_resolver = new CassandraResolver(dns_resolver,
+                                           af,
+                                           30,
+                                           30,
+                                           9160);
+
+    _call_list_store = new CallListStore::Store();
+    _call_list_store->configure_connection(cassandra, 9160, _cass_comm_monitor, _cass_resolver);
+
+    _memento = new MementoAppServer(memento_prefix,
+                                    _call_list_store,
+                                    opt.home_domain,
+                                    max_call_list_length,
+                                    memento_threads,
+                                    call_list_ttl,
+                                    stack_data.stats_aggregator,
+                                    cass_target_latency_us,
+                                    opt.max_tokens,
+                                    opt.init_token_rate,
+                                    opt.min_token_rate,
+                                    opt.max_token_rate,
+                                    exception_handler,
+                                    http_resolver,
+                                    memento_notify_url);
+
+    _memento_sproutlet = new SproutletAppServerShim(_memento,
+                                                    memento_port,
+                                                    memento_uri,
+                                                    incoming_sip_transactions_tbl,
+                                                    outgoing_sip_transactions_tbl);
+    sproutlets.push_back(_memento_sproutlet);
+  }
+  else
+  {
+    TRC_STATUS("Memento plugin disabled");
+  }
+
+  return plugin_loaded;
+}
+
+/// Unloads the Memento plug-in.
+void MementoPlugin::unload()
+{
+  delete _memento_sproutlet;
+  delete _memento;
+  delete _cass_resolver;
+  delete _call_list_store;
+  delete _cass_comm_monitor;
+}
```

Comment:
You've moved all the uts from src/ut to ut. I assume this is to match up with Houdini?

Reply:
Yes, and also ensures compatibility with gcov/build-infra.